### PR TITLE
Update test doc

### PR DESCRIPTION
**Pull request by smlambert**
Add /doc folder
Refactor main README to include images and move User Guide to separate file.

### CODE REVIEW

File: doc/userGuide.md [hunk 1]
```diff
@@ -0,0 +1,182 @@
...
+	* Create Pipeline test build job using Pipeline script from SCM :https://github.com/AdoptOpenJDK/openjdk-tests.git
+	* Configure your sdk build to trigger your created test build
+		* If using execute shell use Parameterized Trigger Plugin to passing in parameters including your sdk job name and job number
+		* If using pipeline update pipeline using following code : 
+
+```
+build job: 'test_build_name', parameters: [string(name: 'PLATFORM', value: '...'), string(name: 'JAVA_VERSION', value: '...'), ... string(name: 'UPSTREAM_JOB_NAME', value: "your sdk job name"), string(name: 'UPSTREAM_JOB_NUMBER', value: "your sdk job number")]
+```
+
+## Local setup and running (via make targets on the command line)
```
Comment (llxia) [inline doc/userGuide.md]: Some of the information mentioned below is about how to use TestKitGen. I think we should redirect user to https://github.com/eclipse/openj9/blob/master/test/README.md, so we can avoid to maintain two files. For now, I think we can just add a link to openj9 test readme.

**Comment by llxia** [comment]
The rest looks good to me. :)

**Comment by llxia** [comment]
Please add Signed-off-by in commit msg. Thanks.

**Comment by smlambert** [comment]
@llxia - I have made some additional changes to the User Guide, can you please re-review?

**Pull request merged**